frontend/lib/wallets/MetaMask.ts

```diff
@@ -21,7 +23,13 @@ export default class MetaMask extends AbstractWallet {
         if (!window.ethereum) {
             return false;
         }
-        if (typeof window.ethereum?.isConnected !== 'function') {
+        if (window.ethereum.providers?.length) {
+            const metaMask = window.ethereum.providers.find((provider: any) => provider.isMetaMask);
+            if (metaMask) {
+                return metaMask.isConnected();
+            }
+        }
+        if (typeof window.ethereum.isConnected !== 'function') {
```

Comment:
`isConnected` function purpose is to check whether the website was previously connected to the metamask, so on the page refresh we can reconnect to it. It prevents users who previously haven't connected their metamask seeing a new dialog as soon a they open the page.

Currently, you're using `window.ethereum.isConnected` which wouldn't be available in case coinbase extension is installed. So my recommendation would be to get (static) provider here first and then check if this function is there.